UChicago VuFind.js

```diff
@@ -0,0 +1,794 @@
+{
+	"translatorID": "f9052879-ab4c-4056-84b4-e963dd98cb5d",
+	"label": "UChicago VuFind",
+	"creator": "Matt Teichman",
+	"target": "^https?://([^/]+\\.)?lib\\.uchicago\\.edu/",
+	"minVersion": "5.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2022-07-06 17:32:33"
+}
+
+// MARC retrieval code: run the MARC import translator, then perform a
+// few adjustments to the output by looking things up in the MARC
+// record
+
+// scrapeMARC function, overall design based on Finna translator
+const scrapeMARC = doc => url => {
```

Comment:
Very Haskell :)

There's nothing technically incorrect about using currying like this, but it's a bit idiosyncratic for this codebase/JS in general. I haven't spotted any when quickly looking over the code, but are there any places here where we're taking advantage of e.g. partially applying these functions? If not, I think I'd argue that we should replace curried functions with traditional argument lists (`doc => url =>` -> `(doc, url) =>`) for clarity/ease of maintenance by others with a JavaScript background.

Reply:
Guilty as charged, your honor.  OCaml and Haskell are my main languages, and I also teach Haskell!  

The one place in the code where I make use of the fact that `scrapeMARC` is curried is in line 213:

https://github.com/zotero/translators/pull/2846/files#diff-0b26c1ec34be25953bbeb714f32ed65fd5d1a2ede776aa3b1eaa8a03c1910064R213

If you think it would be easier for your team to work with, I'd be happy to uncurry `scrapeMARC` and change line 213 around so that it didn't rely on partial application; just let me know what your preference is.